### PR TITLE
Added edge case checking in isGreen

### PR DESCRIPTION
**Pull request by swang392**
Relates to #76700

**Overview**: One edge case not accounted for in the original logic of `isGreen` was for commits with no workflow checks. Similarly, if any of the required checks are not present (ex: if all of the pull checks are skipped), the workflow should not be promoteble. A commit should only be promoteable if there is it least one workflow check from each required group present (i.e. none of them are skipped)

**Test Plan:** Verify that commits on the HUD with no workflow checks are not considered promote-able. Added a test case with no workflows in `test_print_latest_commits.py`

### CODE REVIEW

**Review by seemethere — APPROVED** [review]

File: .github/scripts/print_latest_commits.py [hunk 1]
```diff
@@ -72,6 +72,9 @@ def get_commit_results(commit: str, results: Dict[str, Any]) -> List[Dict[str, A
 def isGreen(commit: str, results: Dict[str, Any]) -> Tuple[bool, str]:
     workflow_checks = get_commit_results(commit, results)
 
+    if len(workflow_checks) == 0:
```
Comment (janeyx99) [inline .github/scripts/print_latest_commits.py:75]: Could we restructure this code to make sure that there is at least one check run for each of the items in: 
     "pull+",
    "^trunk+",
    "^lint+",
    "^linux-binary+",
    "^android-tests+",
    "^windows-binary+"

File: .github/scripts/print_latest_commits.py [hunk 2]
```diff
@@ -5,14 +5,14 @@
...
 
-regex = [
-    "^pull+",
-    "^trunk+",
-    "^lint+",
-    "^linux-binary+",
-    "^android-tests+",
-    "^windows-binary+"
-]
+regex = {
```
Comment (janeyx99) [inline .github/scripts/print_latest_commits.py:8]: The move may be to move this back to a local variable under isGreen as this dictionary changes now.

**Pull request closed without merging**